global/scripts/MyelinMap_BC.sh

```diff
@@ -0,0 +1,172 @@
+opts_SetScriptDescription "implements CIFTI-based Myelin bias fields from native mesh to 32k resolution then back to native mesh; by default changing the group reference's mean with individual map's mean"
+
+#mandatory
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
+opts_AddMandatory '--subject' 'Subject' '100206' "one subject ID"
+opts_AddMandatory '--registration-name' 'RegName' 'MSMAll' "the registration string corresponding to the input files, e.g. 'MSMAll' or 'MSMSulc'"
+opts_AddMandatory '--msm-all-templates' 'MSMAllTemplates' 'path' "path to directory containing MSM All template files, e.g. 'YourFolder/global/templates/MSMAll'"
+#optional inputs
+opts_AddOptional '--use-ind-mean' 'UseIndMean' 'YES or NO' "whether to use the mean of the subject's myelin map as reference map's myelin map mean , defaults to 'YES'" 'YES'
+opts_AddOptional '--mcsigma' 'CorrectionSigma' 'number' "myelin map bias correction sigma, default '$defaultSigma'" "$defaultSigma"
+opts_AddOptional '--myelin-target-file' 'MyelinTarget' 'string' "alternate myelin map target, relative to the --msm-all-templates folder" 'Q1-Q6_RelatedParcellation210.MyelinMap_BC_MSMAll_2_d41_WRN_DeDrift.32k_fs_LR.dscalar.nii'
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+# Verify required environment variables are set and log value
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
+
+#display HCP Pipeline version
+log_Msg "Showing HCP Pipelines version"
+"${HCPPIPEDIR}"/show_version --short
+
+#display the parsed/default values
+opts_ShowValues
+
+# ------------------------------------------------------------------------------
+#  Main processing of script.
+# ------------------------------------------------------------------------------
+log_Msg "Starting main functionality"
+Caret7_Command=${CARET7DIR}/wb_command
+UseIndMeanBool=$(opts_StringToBool "$UseIndMean")
+# default folders
+SubjFolder=${StudyFolder}/${Subject}
+log_Msg "SubjFolder: $SubjFolder"
+AtlasSpaceFolder=${SubjFolder}/MNINonLinear
+log_Msg "AtlasSpaceFolder: $AtlasSpaceFolder"
+T1wFolder=${SubjFolder}/T1w
+log_Msg "T1wFolder: $T1wFolder"
+NativeFolder=${AtlasSpaceFolder}/Native
+log_Msg "NativeFolder: $NativeFolder"
+NativeT1wFolder=${T1wFolder}/Native
+log_Msg "NativeT1wFolder: $NativeT1wFolder"
+# MSMAll templates Myelin Target which is the group average MyelinMap_BC
+MyelinTarget="${MSMAllTemplates}/${MyelinTarget}"
+# 32k setting
+LowResMesh=32
+LowResMeshString=${LowResMesh}k_fs_LR
+LowResFolder=${AtlasSpaceFolder}/fsaverage_LR${LowResMesh}k
+LowResT1wFolder=${T1wFolder}/fsaverage_LR${LowResMesh}k
+LowResBCMapName="atlas_MyelinMap"
+
+# atlas group average reference map
+tempfiles_create atlas_MyelinMap_BC_XXXXXX.dscalar.nii ReferenceMap
+cp "${MyelinTarget}" "$ReferenceMap"
+
+case "$RegName" in
+	(MSMSulc)
```

Comment:
I would advise `(MSMSulc|"")`, since we treat `NONE` or empty string for RegName as referring to MSMSulc in some pipelines ("MSMSulc" is almost never how the pipelines options select that registration).